js/controller.js

```diff
@@ -65,6 +66,38 @@
             refreshMirrorData();
             $interval(refreshMirrorData, 3600000);
 
+            var refreshTrafficData = function() {
+              TrafficService.getTravelDuration().then(function() {
+                var information = TrafficService.getCurrentTime();
+                if (information.hasOwnProperty("error")) {
```

Comment:
Ideally all of this logic should be contained inside the traffic service. There's a lot here that doesn't have to do with display logic.